python/main.py

```diff
@@ -64,24 +75,53 @@ def hello():
 class AddItemResponse(BaseModel):
     message: str
 
+class Item(BaseModel):
+    name: str
+    category: str
+    image_name: str
+
+def insert_item(item: Item):
+    # STEP 4-1: add an implementation to store an item
+    try:
+        with open("items.json", "r") as f:
+            data = json.load(f)
+    except (FileNotFoundError, json.JSONDecodeError):
+        data = {"items": []}
+
+        # Append new item
+    data["items"].append({"name": item.name, "category": item.category, "image_name": item.image_name})
```

Comment:
[nits]
You can get a dictionary more easily.
```suggestion
    data["items"].append(item.model_dump())
```